applications/sdp/mspi/src/main.c

```diff
@@ -0,0 +1,205 @@
+		nrf_vpr_csr_vio_out_set(PIN_OUT_HIGH_MASK(SCLK_PIN));
+		break;
+	}
+	}
+
+	nrf_vpr_csr_vio_config_set(&vio_config);
+}
+
+void prepare_and_send_data(uint8_t *data, uint8_t data_length)
+{
+	/* this is here only temporarly to set command and address parameters */
+	data_buffer[FRAME_COMMAND_IDX] = 0xe5b326c1;
+	data_buffer[FRAME_ADDRESS_IDX] = 0xaabbccdd;
+
+	memcpy(&(data_buffer[2]), data, data_length);
+
+	/* Send command */
+	ce_hold = true;
+	word_size = mspi_dev_configs.cmd_length;
+	data_len = 1;
+	data_to_send = data_buffer;
+
+	if (mspi_dev_configs.io_mode == MSPI_IO_MODE_QUAD) {
+		nrf_vpr_clic_int_pending_set(NRF_VPRCLIC, VEVIF_IRQN(HRT_VEVIF_IDX_WRITE_QUAD));
+	} else {
+		nrf_vpr_clic_int_pending_set(NRF_VPRCLIC, VEVIF_IRQN(HRT_VEVIF_IDX_WRITE_SINGLE));
+	}
+
+	/* Send address */
+	word_size = mspi_dev_configs.addr_length;
+	data_to_send = data_buffer + FRAME_ADDRESS_IDX;
+
+	if (mspi_dev_configs.io_mode == MSPI_IO_MODE_SINGLE ||
+	    mspi_dev_configs.io_mode == MSPI_IO_MODE_QUAD_1_1_4) {
+		nrf_vpr_clic_int_pending_set(NRF_VPRCLIC, VEVIF_IRQN(HRT_VEVIF_IDX_WRITE_SINGLE));
+	} else {
+		nrf_vpr_clic_int_pending_set(NRF_VPRCLIC, VEVIF_IRQN(HRT_VEVIF_IDX_WRITE_QUAD));
+	}
+
+	/* Send data */
+	ce_hold = false;
+	word_size = 32;
+	/* TODO: this system needs to be fixed as for now,
+	 * there are problems when (data_length%4) != 0
+	 */
+	data_len = WORDS_FOR_BYTES(data_length);
+	data_to_send = data_buffer + FRAME_DATA_IDX;
+
+	if (mspi_dev_configs.io_mode == MSPI_IO_MODE_SINGLE) {
+		nrf_vpr_clic_int_pending_set(NRF_VPRCLIC, VEVIF_IRQN(HRT_VEVIF_IDX_WRITE_SINGLE));
+	} else {
+		nrf_vpr_clic_int_pending_set(NRF_VPRCLIC, VEVIF_IRQN(HRT_VEVIF_IDX_WRITE_QUAD));
+	}
+}
+
+__attribute__((interrupt)) void hrt_handler_write_single(void)
+{
+	uint8_t ce_enable_state = (mspi_dev_configs.ce_polarity == MSPI_CE_ACTIVE_LOW) ? 0 : 1;
+
+	write_single_by_word(data_to_send, data_len, counter_top, word_size, ce_enable_state,
+			     ce_hold);
+}
+
+__attribute__((interrupt)) void hrt_handler_write_quad(void)
+{
+	uint8_t ce_enable_state = (mspi_dev_configs.ce_polarity == MSPI_CE_ACTIVE_LOW) ? 0 : 1;
+
+	write_quad_by_word(data_to_send, data_len, counter_top, word_size, ce_enable_state,
+			   ce_hold);
+}
+
+int main(void)
+{
+	IRQ_DIRECT_CONNECT(HRT_VEVIF_IDX_WRITE_SINGLE, HRT_IRQ_PRIORITY, hrt_handler_write_single,
+			   0);
+	nrf_vpr_clic_int_enable_set(NRF_VPRCLIC, VEVIF_IRQN(HRT_VEVIF_IDX_WRITE_SINGLE), true);
+
+	IRQ_DIRECT_CONNECT(HRT_VEVIF_IDX_WRITE_QUAD, HRT_IRQ_PRIORITY, hrt_handler_write_quad, 0);
+	nrf_vpr_clic_int_enable_set(NRF_VPRCLIC, VEVIF_IRQN(HRT_VEVIF_IDX_WRITE_QUAD), true);
+
+	nrf_vpr_csr_rtperiph_enable_set(true);
+
+	nrf_gpio_pin_dir_t dir = NRF_GPIO_PIN_DIR_OUTPUT;
+	nrf_gpio_pin_input_t input = NRF_GPIO_PIN_INPUT_DISCONNECT;
+	nrf_gpio_pin_pull_t pull = NRF_GPIO_PIN_NOPULL;
+	nrf_gpio_pin_drive_t drive = NRF_GPIO_PIN_E0E1;
+
+	nrfy_gpio_reconfigure(NRF_GPIO_PIN_MAP(2, SCLK_PIN), &dir, &input, &pull, &drive, NULL);
```

Comment:
Same as `nrfy_gpio_pin_control_select` if this is supposed to be done on APP side.